numba/core/ir_utils.py

```diff
@@ -851,6 +853,12 @@ def find_potential_aliases(blocks, args, typemap, func_ir, alias_map=None,
                         _add_alias(lhs, expr.args[0].name, alias_map, arg_aliases)
                     if isinstance(fmod, ir.Var) and fname in np_alias_funcs:
                         _add_alias(lhs, fmod.name, alias_map, arg_aliases)
+                    if isinstance(lhstype, types.containers.BaseTuple):
```

Comment:
Shouldn't this be limited to namedtuple construction only instead of any function call returning any type of tuple?

Reply:
I considered this issue but came to the conclusion that any function given a value and returning a tuple could put that value in the tuple.  I believe this is consistent with our general approach which has been to assume the worst case at function calls (we don't try to recursively analyze those calls to get a more accurate result right?).  I don't think it matters that in this case it is a namedtuple and in fact I think maybe this is not even generic enough because there are other containers in this module for which these rules might possibly apply.

Reply:
Ok, makes sense. Looks like this is mostly used for dead code elimination and wouldn't affect common cases anyways.

The only other use is in array analysis with `get_canonical_alias`, where this change can potentially cause problems. For example, `A, B = f(C)` assigns aliases for A, B, C and the information is used to replace `A.attr` with `C.attr`. Am I understanding this correctly? If so, our "maybe alias" is used as "definitely alias" in this context which is problematic (a problem that the existing code has, but this PR makes it worse).

Reply:
If indeed what we have which is "maybe alias" is used a "definitely alias" then that is a problem.